core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/upgrade/DistroXUpgradeAvailabilityService.java

```diff
@@ -71,6 +74,7 @@ public UpgradeV4Response checkForUpgrade(NameOrCrn nameOrCrn, Long workspaceId,
     }
 
     private List<ImageInfoV4Response> filterCandidates(String accountId, Stack stack, UpgradeV4Request request, UpgradeV4Response upgradeV4Response) {
+        filterOnlyPatchUpgradesIfRuntimeUpgradeDisabled(accountId, stack.getName(), upgradeV4Response);
```

Comment:
Now I checked the code better and just wonder why actually stackOperations.checkForClusterUpgrade is not returning maintenance upgrade candidates.